main_GANs.py

```diff
@@ -55,29 +66,58 @@
     epochs = int(arguments['--epochs'])
     optimizer_name = arguments['--optimizer']
     learning_rate = float(arguments['--learning_rate'])
-
-    model = MLP_GANs_model(mnist_data(rand_rotation = False, max_degree = 90))
-    model.train(num_epochs = epochs, lr_x = torch.tensor([learning_rate]), lr_y = torch.tensor([learning_rate]),
-                optimizer_name = optimizer_name, verbose = True, label_smoothing = False, single_number = 4) # save_path = ''
-
-    #model = DCGANs_model(mnist_data_dcgans(rand_rotation = False, max_degree = 90))
-    #model.train(num_epochs = epochs, lr_x = torch.tensor([learning_rate]), lr_y = torch.tensor([learning_rate]),
-#                optimizer_name = optimizer_name, verbose = True, label_smoothing = False, single_number = 9) #save_path = ''
-
-    #model.save_models()
-
-    plt.figure()
-    plt.plot([x for x in range(0, len(model.D_error_real_history))],
-             model.D_error_real_history)
-    plt.plot([x for x in range(0, len(model.D_error_fake_history))],
-             model.D_error_fake_history)
-    plt.plot([x for x in range(0, len(model.G_error_history))],
-             model.G_error_history)
-    plt.xlabel('Iterations')
-    plt.ylabel('Loss function value')
-    plt.legend([
-        'Discriminator: Loss on Real Data', 'Discriminator: Loss on Fake Data',
-        'Generator: Loss'
-    ])
-    plt.savefig('cost_report.png')
-
+    model_switch = arguments['--model']
+
+    if model_switch == 'MLP':
+        print("Using MLP implementation of GANs: MLP_GANs_model")
+        model = MLP_GANs_model(mnist_data(rand_rotation=True, max_degree=90))
```

Comment:
Hmm, why was `rand_rotation` changed? Does it really belong to this PR?

Reply:
No :)